tensorly/backend/core.py

```diff
@@ -608,10 +608,14 @@ def kr(self, matrices):
             s3, s4 = self.shape(e)
             if not s2 == s4 == n_col:
                 raise ValueError('All matrices should have the same number of columns.')
+
             a = self.reshape(res, (s1, 1, s2))
             b = self.reshape(e, (1, s3, s4))
             res = self.reshape(a * b, (-1, n_col))
-        return res
+        
+        m = self.reshape(mask, (-1, 1)) if mask is not None else 1
+        
+        return res*m
```

Comment:
Did this change? I thought the mask had to be incorporated into the computation in order to stay sparse safe in some cases.

Reply:
That was the whole point of making `mask` an argument of the function, rather than having the caller multiply the result.

Reply:
Yes, true, if this is sparse safe, `mask` doesn't need to be in the khatri-rao / kronecker.
Why would it not be sparse-safe -- It's just an element-wise multiplication with broadcasting?

Reply:
I think there were cases where the kr itself was large but `kr*mask` was sparse (`mask` is assumed to be sparse). In the parafac code, doesn't it build the full tensor from the factors?

Reply:
We changed that part -- instead of using the khatri-rao we use ``tensorly.tenalg. unfolding_dot_khatri_rao`` so shouldn't be an issue in CP. Actually might be worth pulling out of the function for kr below and either put it back in the computation here (I think you had added this [here](https://github.com/tensorly/tensorly/pull/91/files/429b4a767b8e4085763b2da758a934d4012fd8b3#r265118645)  ).

Reply:
I don't remember all the details of this. If the mask example notebook works then I guess the algorithm is working fine.